src/python/grpcio/grpc/_auth.py

```diff
@@ -60,15 +69,7 @@ def __call__(self, context, callback):
         else:
             future = self._pool.submit(self._credentials.get_access_token)
         future.add_done_callback(
-            lambda x: self._get_token_callback(callback, x))
-
-    def _get_token_callback(self, callback, future):
-        try:
-            access_token = future.result().access_token
-        except Exception as e:
-            _sign_request(callback, None, e)
-        else:
-            _sign_request(callback, access_token, None)
+            lambda ignored_future: _get_token_callback(callback, future))
```

Comment:
I'm not sure I understand, but I'll say that this code is going to look strange no matter the circumstances because of the way that `add_done_callback` has semantics of "the value passed to your behavior will be an object to which we already know you have access". Can you sketch out the shape of the implementation that you think might be more sensible?